plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/CDKConstantUtility.java

```diff
@@ -10,29 +10,143 @@
  ******************************************************************************/ 
 package org.jboss.tools.openshift.cdk.server.core.internal;
 
-import org.eclipse.debug.core.ILaunchConfiguration;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.debug.core.ILaunchConfiguration;
 import org.eclipse.wst.server.core.IServer;
 import org.jboss.tools.openshift.cdk.server.core.internal.adapter.controllers.IExternalLaunchConstants;
 
-// TODO - allow customization of this location
 public class CDKConstantUtility {
 	private static final String VAGRANT_LOCATION_LINUX = "/usr/bin/vagrant";
-
+	
+	// Seems weird?  See https://github.com/mitchellh/vagrant/issues/1652
+	private static final String VAGRANT_LOCATION_WINDOWS = "C:\\HashiCorp\\Vagrant\\bin\\vagrant.exe";
+	
+	// Linux command to find vagrant full path
+	private static final String LINUX_WHICH_VAGRANT = "which vagrant";
+	
+	// Windows command to find vagrant
+	private static final String WINDOWS_WHERE_VAGRANT = "where vagrant";
+	
+	
+	private static String discoveredVagrantLocation = null;
+	private static boolean searchFailed = false;
+	
+	
 	public static String getVagrantLocation() {
-		return VAGRANT_LOCATION_LINUX;
+		return findVagrantLocation();
```

Comment:
There are times we call out via Runtime.exec() in which case the existing path is in control.  But there are also times we use the external tools launch configuration, in which case we need to know the full location.  In general, we use the launch config when we want a console to appear with the output, and we use Runtime.exec() when we just want it done in the background.